### PR TITLE
Spring Boot Hazelcast integration example

### PR DESCRIPTION
**Pull request by gAmUssA**
Added Spring Boot Hazelcast integration example 

@mesutcelik @mustafaiman, please, review and merge

### CODE REVIEW

File: hazelcast-integration/springboot-caching/.gitignore [hunk 1]
```diff
@@ -0,0 +1,65 @@
+
```
Comment (mustafaiman) [inline hazelcast-integration/springboot-caching/.gitignore]: I think we do not introduce seperate `.gitignore` files with new samples. Is this neccessary?

File: hazelcast-integration/springboot-caching/src/main/resources/banner.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+ [0;1;34;94m_[0m   [0;1;34;94m_[0m               [0;34m_[0m               [0;37m_[0m        [0;37m__[0;1;30;90m____[0m             [0;1;34;94m_[0m
```
Comment (mustafaiman) [inline hazelcast-integration/springboot-caching/src/main/resources/banner.txt]: What is this file?

**Comment by mustafaiman** [comment]
+1 (minor comments). Nice sample!

File: hazelcast-integration/springboot-caching/.gitignore (hunk 1 above)
Comment (gAmUssA) [inline hazelcast-integration/springboot-caching/.gitignore]: @mustafaiman Local (in folder only) ignore file. Will not affect rest of the projects.

File: hazelcast-integration/springboot-caching/src/main/resources/banner.txt (hunk 2 above)
Comment (gAmUssA) [inline hazelcast-integration/springboot-caching/src/main/resources/banner.txt]: @mustafaiman colorful asciiart banner

File: hazelcast-integration/springboot-caching/src/main/resources/hazelcast-client.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  ~  limitations under the License.
+  ~
+  -->
+<hazelcast-client
+        xsi:schemaLocation="http://www.hazelcast.com/schema/client-config http://www.hazelcast.com/schema/client-config/hazelcast-client-config-3.6.xsd"
+        xmlns="http://www.hazelcast.com/schema/client-config"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+
+    <group>
+        <name>bootifulclister</name>
```
Comment (emrahkocaman) [inline hazelcast-integration/springboot-caching/src/main/resources/hazelcast-client.xml]: I think there is a typo here

File: hazelcast-integration/springboot-caching/src/main/resources/hazelcast.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  ~
+  -->
+
+<hazelcast
+        xsi:schemaLocation="http://www.hazelcast.com/schema/config http://www.hazelcast.com/schema/config/hazelcast-config-3.6.xsd"
+        xmlns="http://www.hazelcast.com/schema/config"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+
+    <group>
+        <name>bootifulclister</name>
```
Comment (emrahkocaman) [inline hazelcast-integration/springboot-caching/src/main/resources/hazelcast.xml]: typo?

File: hazelcast-integration/springboot-caching/src/main/resources/hazelcast-client.xml (hunk 3 above)
Comment (gAmUssA) [inline hazelcast-integration/springboot-caching/src/main/resources/hazelcast-client.xml]: fixed @emrahkocaman

File: hazelcast-integration/springboot-caching/src/main/resources/hazelcast.xml (hunk 4 above)
Comment (gAmUssA) [inline hazelcast-integration/springboot-caching/src/main/resources/hazelcast.xml]: fixed @emrahkocaman

File: hazelcast-integration/springboot-caching/.gitignore (hunk 1 above)
Comment (mustafaiman) [inline hazelcast-integration/springboot-caching/.gitignore]: I know that but it seemed redundant to me. Does this project create any file that should be ignored and is not ignored by root `.gitignore`? If this is the case you can put only those files in `.gitignore`.

**Comment by emrahkocaman** [comment]
:+1:

File: hazelcast-integration/springboot-caching/.gitignore (hunk 1 above)
Comment (gAmUssA) [inline hazelcast-integration/springboot-caching/.gitignore]: @mustafaiman It's a matter of taste. Just want to have everything I need in place. I prefer don't mess with global things in a local commit. Do you think it does make sense to move those ignores to global (project wise) config (in separate PR)?

Comment (gAmUssA) [inline hazelcast-integration/springboot-caching/.gitignore]: fixed @mustafaiman

File: hazelcast-integration/springboot-caching/src/main/resources/banner.txt (hunk 2 above)
Comment (gAmUssA) [inline hazelcast-integration/springboot-caching/src/main/resources/banner.txt]: @mustafaiman here is a demo https://asciinema.org/a/32526

**Pull request merged**